packages/transaction-controller/src/TransactionController.ts

```diff
@@ -556,14 +621,117 @@ export class TransactionController extends BaseControllerV1<
 
     this.addPendingTransactionTrackerListeners();
 
+    // when transactionsController state changes
+    // check for pending transactions and start polling if there are any
+    this.subscribe(this.#checkForPendingTransactionAndStartPolling);
+
+    // TODO once v2 is merged make sure this only runs when
+    // selectedNetworkClientId changes
     onNetworkStateChange(() => {
       log('Detected network change', this.getChainId());
+      this.pendingTransactionTracker.onStateChange();
       this.onBootCleanup();
     });
 
     this.onBootCleanup();
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
+      (_, patches) => {
+        if (this.enableMultichain) {
+          const networkClients = this.getNetworkClientRegistry();
+          patches.forEach(({ op, path }) => {
+            if (op === 'remove' && path[0] === 'networkConfigurations') {
+              const networkClientId = path[1] as NetworkClientId;
+              delete networkClients[networkClientId];
+            }
+          });
+
+          this.#refreshTrackingMap(networkClients);
+          this.#refreshEtherscanRemoteTransactionSources(networkClients);
+        }
+      },
+    );
+    if (this.enableMultichain) {
+      this.#initTrackingMap();
+    }
   }
 
+  setEnableMultichain(enableMultichain: boolean) {
+    this.enableMultichain = enableMultichain;
+    if (enableMultichain) {
+      this.#initTrackingMap();
+    }
+  }
+
+  /**
+   * Stops polling and removes listeners to prepare the controller for garbage collection.
+   */
+  destroy() {
+    this.#stopAllTracking();
+  }
+
+  #refreshEtherscanRemoteTransactionSources = (
+    networkClients: ReturnType<NetworkController['getNetworkClientRegistry']>,
+  ) => {
+    // this will be prettier when we have consolidated network clients with a single chainId:
+    // check if there are still other network clients using the same chainId
+    // if not remove the etherscanRemoteTransaction source from the map
+    const chainIdsInRegistry = new Set();
```

Comment:
Since there is a few types and methods supporting the tracking map, in the interest of modularity, does it warrant its own helper to encapsulate all the logic and perhaps simplify unit tests also?